src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java

```diff
@@ -183,17 +186,20 @@ public ResponseEntity adjustStock(@PathVariable long facilityId,
                 return OpenLmisResponse.error(messageService.message("error.product.unknown"), HttpStatus.BAD_REQUEST);
 
             // validate reason
-            StockAdjustmentReason reason = stockAdjustmentReasonRepository.getAdjustmentReasonByName(
-                event.getReasonName());
-            if(null == reason)
-                return OpenLmisResponse.error(messageService.message("error.stockadjustmentreason.unknown"),
-                    HttpStatus.BAD_REQUEST);
+            StockAdjustmentReason reason = null;
+            if (StockEventType.ADJUSTMENT == event.getType()) {
+                reason = stockAdjustmentReasonRepository.getAdjustmentReasonByName(
+                        event.getReasonName());
+                if(null == reason)
+                    return OpenLmisResponse.error(messageService.message("error.stockadjustmentreason.unknown"),
+                            HttpStatus.BAD_REQUEST);
+            }
 
             // get or create stock card
             //TODO:  this call might create a stock card if it doesn't exist, need to implement permission check
             StockCard card = service.getOrCreateStockCard(facilityId, productId);
             if(null == card)
-                return OpenLmisResponse.error("Unable to adjust stock for facility and product",
+                return OpenLmisResponse.error("Unable to process stock for facility and product",
                     HttpStatus.BAD_REQUEST);
```

Comment:
I don’t think this is a BAD_REQUEST. Rather, a null StockCard at this point indicates a server-side error (code 500).